Tests/BSONTests/ExtendedJSONConversionTests.swift

```diff
@@ -157,11 +165,33 @@ open class ExtendedJSONConversionTestCase: BSONTestCase {
     }
 
     func testCodeWScope() throws {
-        // Skipping until there is a BSON.init(fromExtJSON) because its recursive
+        // Success case
+        expect(try BSONCodeWithScope(fromExtJSON: ["$code": "javascript", "$scope": ["doc": "scope"]], keyPath: []))
+            .to(equal(BSONCodeWithScope(
+                code: "javascript",
+                scope: BSONDocument(keyValuePairs: [("doc", BSON.string("scope"))])
```

Comment:
you can initialize this directly from an object literal like you do with `JSON` (e.g. `["doc": "scope"]`)